src/apiron/endpoint/endpoint.py

```diff
@@ -1,25 +1,44 @@
 import logging
 import string
+import sys
 import warnings
 from functools import partial, update_wrapper
-from typing import Any, Dict, Iterable, List, Union
+from typing import Any, Callable, Dict, Iterable, List, TypeVar, Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 10):
+        from typing import Concatenate, ParamSpec
+    else:
+        from typing_extensions import Concatenate, ParamSpec
```

Comment:
Although mypy installs it as an abstract dependency, let's add `typing_extensions` to the `[testenv:typecheck]deps` since it's now a concrete dependency.